lib/oneview-sdk/resource/api500/c7000/storage_pool.rb

```diff
@@ -40,6 +40,29 @@ def delete(*)
           unavailable_method
         end
 
+        # Retrieve resource details based on this resource's name or URI.
+        # @note Name or URI must be specified inside the resource
+        # @return [Boolean] Whether or not retrieve was successful
+        def retrieve!
+          raise IncompleteResource, 'Must set resource name or uri before trying to retrieve!' unless @data['name'] || @data['uri']
+          if @data['name'] && @data['storageSystemUri']
+            results = self.class.find_by(@client, name: @data['name'], storageSystemUri: @data['storageSystemUri'])
+            if results.size == 1
+              set_all(results[0].data)
+              return true
+            end
+          end
+          super
```

Comment:
Is there a point to calling `super` when we've already found the resource?
How about moving this to [line 48](https://github.com/HewlettPackard/oneview-sdk-ruby/pull/270/files#diff-c40ca7a34417c89fafffff2e3e5bfd96R48) as:
```ruby
return super if @data['uri']
```
After that you can assume that if you get past that code:
1- `@data['name'] && @data['storageSystemUri']` **are** both set 
2- `@data['uri']` **was not** set

Therefore you can go ahead and drop the `if @data['name'] && @data['storageSystemUri']` line.

Looks like that'd be cleaner, unless we want to be calling `super` in these cases for some reason I'm missing